### PR TITLE
feature: add name to all operators

### PR DESCRIPTION
**Pull request by mrava87**
Add an optional parameter `name` to every linear operator to be used in describe method. Addresses the last main points of discussion in https://github.com/PyLops/pylops/issues/293

### CODE REVIEW

**Comment by mrava87** [comment]
@cako when you have time check the comments in the issue and this PR :)

File: pylops/basicoperators/CausalIntegration.py [hunk 1]
```diff
@@ -96,6 +98,7 @@ def __init__(
         dtype="float64",
         kind="full",
         removefirst=False,
+        name="C",
```
Comment (cako) [inline pylops/basicoperators/CausalIntegration.py:101]: Maybe **I** for causal integration? We already have C for Conj and Convolution.

File: pylops/basicoperators/FunctionOperator.py [hunk 2]
```diff
@@ -65,8 +69,7 @@ def __init__(self, f, *args, **kwargs):
             self.dtype = kwargs["dtype"]
         except KeyError:
             self.dtype = "float64"
-        self.explicit = False
-
+        super().__init__(explicit=False, clinear=True, name=kwargs.get("name", None))
```
Comment (cako) [inline pylops/basicoperators/FunctionOperator.py:72]: Why not provide "\mathrm{Fun}" or "F" as default?

File: pylops/basicoperators/MemoizeOperator.py [hunk 3]
```diff
@@ -29,11 +33,8 @@ class MemoizeOperator(LinearOperator):
 
     """
 
-    def __init__(self, Op, max_neval=10):
-        self.Op = Op
-        self.shape = Op.shape
-        self.dtype = np.dtype(Op.dtype)
-        self.explicit = False
+    def __init__(self, Op, max_neval=10, name="M"):
+        super().__init__(Op=Op, explicit=False, clinear=None, name=name)
```
Comment (cako) [inline pylops/basicoperators/MemoizeOperator.py:37]: The memoized operator should probably just inherit the name of the original operator.

**Review by cako — APPROVED** [review]
I am happy with the changes, just have a look at some of my comments!

File: pylops/basicoperators/FunctionOperator.py (hunk 2 above)
Comment (mrava87) [reply pylops/basicoperators/FunctionOperator.py:72]: "F" is good :)

File: pylops/basicoperators/MemoizeOperator.py (hunk 3 above)
Comment (mrava87) [reply pylops/basicoperators/MemoizeOperator.py:37]: Agree

**Comment by mrava87** [comment]
Build failing likely due to https://github.com/scikit-fmm/scikit-fmm/pull/66 

Let's wait until it's fixed :)

**Comment by jkfurtney** [comment]
I can put a scikit-fmm pypi release out later this afternoon if it will help your project.

**Comment by mrava87** [comment]
> I can put a scikit-fmm pypi release out later this afternoon if it will help your project.

Thanks a lot! Do not rush, we can wait but it will definitely be good to have scikit-fmm working again so we don't need to do some quick fix to push this PR :)

**Comment by mrava87** [comment]
@cako can you have a quick look at the remaining submodules where I added name. So that once CI is back I can merge it :)

**Review by cako — APPROVED** [review]

**Comment by mrava87** [comment]
> OK, I just pushed a scikit-fmm release out with your fix.

Fantastic, thanks a lot! It works now :)

**Pull request merged**